### DISCUSSION TITLE
Would a script for convenient hashtags be useful for KES?

### DISCUSSION THREAD

**Discussion by minnieoo** [aclist/kbin-megamod#69] [Archived]
so i started making this personal script a few weeks ago on and off for myself. its purpose is to allow me to store hashtags i use often nearby, able to be copied, deleted, or edited, so that i dont need to retype them for every post, or open a notepad where i could alternatively keep them. its all for convenience.

features:
- label your tags
- delete, edit, and easily copy your tags
- paste tags (you previously could not paste tags, you must type them individually and click enter. the script automatically separates them)

i am attaching a video to showcase some of it's functionality, but it is unfinished. currently it doesn't store the tags (they will be gone upon leaving the page/refreshing), nor can they be edited, but i am getting to that. 

my question is should I consider adding this to KES when i'm done? would anyone find it useful? or shall i keep it personal?

https://github.com/aclist/kbin-kes/assets/137783738/e8ae560c-8d85-4ad4-9006-5fd97ce5f459

**Comment by aclist** [comment]
That's a crazy script, feels like we are in the early Internet with all these hashtags again!

I don't see a problem with it, KES is just the interface, people can include whatever they want, provided it is reasonable. You never know who else might be dying for this kind of feature.

**Reply by minnieoo** [reply]
hashtags are the key to discovery as ugly as they are lol. im glad kbin keeps them tucked away, they serve their purpose without being loud. i was just thinking since this KES is the main script people would download, that you might have some thoughts about what to and what not to include so it doesnt get crowded with things that wont be useful.

**Reply by aclist** [reply]
It is early days, the more the merrier. We can always chuck your script in the garbage can later if it gets bloated :D  Just kidding. Anyway, it seems reasonable, I'd like to see it further along to get the full picture. It sounds like you need to persistently store the tags, so maybe look into the way KES uses the localStorage object and namespace them under a clearly defined name. But it sounds like the settings are already inside your script, so would it just be a toggle on/off sort of thing?

We don't have a well-defined design logic at the moment for how to handle scripts that are themselves menus, or whether this should even matter. It doesn't cause any problems per se, but something to think about...what if the script has its own internal settings menu? Is this good? Bad? (rhetorical question)
